resqpy/crs.py

```diff
@@ -30,12 +35,23 @@ def resqml_type(self):
    valid_axis_orders = ("easting northing", "northing easting", "westing southing",
                         "southing westing", "northing westing", "westing northing")
 
-   def __init__(self, parent_model, crs_root = None, uuid = None,
-                x_offset = 0.0, y_offset = 0.0, z_offset = 0.0,
-                rotation = 0.0, xy_units = 'm', z_units = 'm',
-                z_inc_down = True, axis_order = 'easting northing',
-                time_units = None, epsg_code = None,
-                title = None, originator = None, extra_metadata = None):
+   def __init__(self,
+                parent_model: 'rq.Model',
+                crs_root = None,  # deprecated
+                uuid: Optional[uuid.UUID] = None,
+                x_offset: Optional[float] = 0.0,
```

Comment:
Is this really optional? The default is `0.0` ; what would `None` mean?

Reply:
Ah, I assumed that Optional means that calling code doesn't need to pass a value, ie. it is a keyword argument with a default value.  Almost all the Crs attributes are required by RESQML. The exceptions, from memory, are time_units (only present when z units are time), epsg code, parent crs